common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/blockconnections/ConnectionData.java

```diff
@@ -73,8 +75,10 @@ public static void update(UserConnection user, Position position) throws Excepti
             }
 
             int newBlockState = handler.connect(user, pos, blockState);
-            if (newBlockState == blockState && blockConnectionProvider.storesBlocks()) {
-                continue;
+            if (newBlockState == blockState) {
+                if (inSync == null) inSync = blockConnectionProvider.storesBlocks(user, position);
```

Comment:
Code style here and in the other cases: expand the ifs with brackets and a new line